main.c

```diff
@@ -280,26 +339,8 @@ inline uint8_t min(uint8_t a, uint8_t b) {
  * Update lighting table as per animation
  */
 void animationCallback(GPTDriver* _driver) {
-  profile currentFunction = profiles[currentProfile];
-  if (currentFunction == animatedRainbowVertical) {
-    gptChangeInterval(_driver, ANIMATION_TIMER_FREQUENCY/5);
```

Comment:
not sure about performance implications of calling `gptChangeInterval` on every call of the callback.
I changed this callback to be called only when switching profiles, switching speeds, disabling/enabling LEDs.

Reply:
Yeah this is a good change. It is somewhat expensive since it holds the chSysLock. (It mostly masks all interrupts) then it just writes a register.